llvm/lib/Target/RISCV/RISCVISelLowering.cpp

```diff
@@ -21906,6 +21906,7 @@ bool RISCVTargetLowering::lowerInterleavedStore(StoreInst *SI,
                                                 ShuffleVectorInst *SVI,
```

Comment:
Just to double check. The caller of this already verified the store is non-volatile and non-atomic? I should have asked that in the load patch.

Reply:
InterleavedAccessImpl::lowerInterleavedStore has an isSimple check at the very top before any of this code gets invoked.